### PR TITLE
Removing metadata but keeping Primary Entry crashes app

### PR DESCRIPTION
**Pull request by antonioalwan**
## Proposed changes

Handling Keychain error by checking errSecSuccess and validating metadata blob before deserializing
Adding fewer tests to capture that scenario.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

## Additional information
https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2575330

### CODE REVIEW

File: IdentityCore/src/workplacejoin/MSIDWPJMetadata.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+@interface MSIDWPJMetadata : NSObject
+
+@property (nonatomic) NSString *certificateThumbprint;
+@property (nonatomic) NSString *cloudHost;
+@property (nonatomic) NSString *deviceID;
+@property (nonatomic) NSString *tenantIdentifier;
+@property (nonatomic) NSString *upn;
+
+- (NSDictionary *)serializeWithFormat:(BOOL)usePrimaryFormat;
```
Comment (kaisong1990) [inline IdentityCore/src/workplacejoin/MSIDWPJMetadata.h:34]: Nit: please consider declaring nullable/non-null. There should be a block when creating the class

File: IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase+Internal.h [hunk 2]
```diff
@@ -40,6 +42,17 @@
                                                  context:(id<MSIDRequestContext>_Nullable)context
                                                    error:(NSError*__nullable*__nullable)error;
 
++ (nullable NSString *)getPrimaryEccTenantWithSharedAccessGroup:(NSString *_Nullable)sharedAccessGroup
+                                                        context:(id <MSIDRequestContext> _Nullable)context
+                                                          error:(NSError *_Nullable *_Nullable)error;
+
+
++ (MSIDWPJMetadata *_Nullable)readWPJMetadataWithSharedAccessGroup:(NSString *_Nullable)sharedAccessGroup
```
Comment (kaisong1990) [inline IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase+Internal.h:50]: nit; Is it possible to keep the same format as `(nullable  MSIDWPJMetadata *)`?

File: IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m [hunk 3]
```diff
@@ -469,31 +437,50 @@ + (MSIDWPJMetadata *)readWPJMetadataWithSharedAccessGroup:(NSString *)sharedAcce
...
+                                                                          error:&subError];
+        if (!decodedDataDict || subError)
+        {
+            if (error)
+            {
+                *error = MSIDCreateError(MSIDKeychainErrorDomain, status, @"WPJ metadata deserialization failed.", nil, nil, subError, context.correlationId, nil, NO);
+            }
+    
+            return nil;
+        }
```
Comment (kaisong1990) [inline IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m:467]: nit: Space after '}'

File: IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase+Internal.h (hunk 2 above)
Comment (antonioalwan) [reply IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase+Internal.h]: addressed

File: IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m (hunk 3 above)
Comment (antonioalwan) [reply IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m:467]: addressed

File: IdentityCore/src/workplacejoin/MSIDWPJMetadata.h (hunk 1 above)
Comment (antonioalwan) [reply IdentityCore/src/workplacejoin/MSIDWPJMetadata.h]: addressed

**Review by oldalton — APPROVED** [review]

**Pull request merged**